### PR TITLE
FEATURE: SD-2656 adding apis to config date format, timezone for Org.

### PR DESCRIPTION
**Pull request by dlcd2203**
Also, we should add the type of settings to filter data, because some settings has dynamic id

JIRA: SD-2656

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12299/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14232/ : SUCCESS in 14m 52s

File: libs/sdk-backend-spi/src/organization/settings/index.ts [hunk 1]
```diff
@@ -26,6 +26,24 @@ export interface IOrganizationSettingsService {
...
     setLocale(locale: string): Promise<void>;
 
+    /**
+     * Sets timezone for organization.
+     *
+     * @param locale - for example "en-US", "cs-CZ", etc.
+     *
+     * @returns promise
+     */
+    setTimeZone(timezoneId: string): Promise<void>;
```
Comment (anvuong119) [inline libs/sdk-backend-spi/src/organization/settings/index.ts:36]: - Should we change the function name `setTimeZone` → `setTimezone`?
  `Timezone` itself is a valid word, there is no need to capitalize the letter `Z`, is there?

- Should we change the param name `timezoneId` → `timezone`?

File: libs/sdk-backend-spi/src/organization/settings/index.ts [hunk 2]
```diff
@@ -26,6 +26,24 @@ export interface IOrganizationSettingsService {
      */
     setLocale(locale: string): Promise<void>;
 
+    /**
+     * Sets timezone for organization.
+     *
+     * @param locale - for example "en-US", "cs-CZ", etc.
```
Comment (anvuong119) [inline libs/sdk-backend-spi/src/organization/settings/index.ts:32]: This description seems to be invalid, doesn't it?

→ Please update this accordingly.

File: libs/sdk-backend-spi/src/organization/settings/index.ts [hunk 3]
```diff
@@ -26,6 +26,24 @@ export interface IOrganizationSettingsService {
...
+    setTimeZone(timezoneId: string): Promise<void>;
+
+    /**
+     * Sets date format for organization.
+     *
+     * @param formatId - example "en-US", "cs-CZ", etc.
+     *
+     * @returns promise
+     */
+    setDateFormat(formatId: string): Promise<void>;
```
Comment (anvuong119) [inline libs/sdk-backend-spi/src/organization/settings/index.ts:45]: Should we change the param name `formatId` → `dateFormat`?

As the function name is `setDateFormat`, it would make more sense if the param is named `dateFormat`.

File: libs/sdk-backend-spi/src/organization/settings/index.ts [hunk 4]
```diff
@@ -26,6 +26,24 @@ export interface IOrganizationSettingsService {
...
+     * @param locale - for example "en-US", "cs-CZ", etc.
+     *
+     * @returns promise
+     */
+    setTimeZone(timezoneId: string): Promise<void>;
+
+    /**
+     * Sets date format for organization.
+     *
+     * @param formatId - example "en-US", "cs-CZ", etc.
```
Comment (anvuong119) [inline libs/sdk-backend-spi/src/organization/settings/index.ts:41]: Please update this line accordingly. Especially the example, as it seems to be irrelevant.

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts [hunk 5]
```diff
@@ -18,6 +18,14 @@ export class OrganizationSettingsService
         return this.setSetting("WHITE_LABELING", whiteLabeling);
     }
 
+    public async setTimeZone(timezoneId: string): Promise<void> {
+        return this.setSetting("TIMEZONE", { value: timezoneId });
+    }
+
+    public async setDateFormat(formatId: string): Promise<void> {
+        return this.setSetting("FORMAT_LOCALE", { value: formatId });
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/organization/settings.ts:26]: Just a confirmation, is it `"FORMAT_LOCALE"` instead of something like `"DATE_FORMAT"`?

File: libs/sdk-backend-base/src/dummyBackend/index.ts [hunk 6]
```diff
@@ -736,6 +738,14 @@ class DummyWorkspaceSettingsService implements IWorkspaceSettingsService {
     setLocale(_locale: string): Promise<void> {
```
Comment (anvuong119) [inline libs/sdk-backend-base/src/dummyBackend/index.ts:738]: Please add an empty line to separate the definition of `getSettingsForCurrentUser` (line 725) from the function above it.

File: libs/sdk-backend-base/src/dummyBackend/index.ts [hunk 7]
```diff
@@ -736,6 +738,14 @@ class DummyWorkspaceSettingsService implements IWorkspaceSettingsService {
...
         return Promise.resolve();
     }
+
+    setTimeZone(_timeZoneId: string): Promise<void> {
```
Comment (anvuong119) [inline libs/sdk-backend-base/src/dummyBackend/index.ts:742]: - Should we change the function name `setTimeZone` → `setTimezone`?
  `Timezone` itself is a valid word, there is no need to capitalize the letter `Z`, is there?

- Should we change the param name `_timeZoneId` → `_timezone`?

File: libs/sdk-backend-base/src/dummyBackend/index.ts [hunk 8]
```diff
@@ -736,6 +738,14 @@ class DummyWorkspaceSettingsService implements IWorkspaceSettingsService {
...
+        return Promise.resolve();
+    }
+
+    setDateFormat(_formatId: string): Promise<void> {
```
Comment (anvuong119) [inline libs/sdk-backend-base/src/dummyBackend/index.ts:746]: Should we change the param name `_formatId` → `_dateFormat`?

As the function name is `setDateFormat`, it would make more sense if the param is named `_dateFormat`.

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts [hunk 9]
```diff
@@ -26,7 +34,7 @@ export class OrganizationSettingsService
         return data.data.reduce((result: ISettings, setting) => {
             return {
                 ...result,
-                [setting.id]: unwrapSettingContent(setting.attributes?.content),
+                [setting.id]: unwrapSettingContent(setting.attributes),
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/organization/settings.ts:37]: Just a confirmation: do we really need to make this change? Is it because the function `getSettings` does not work as expected?

Comment (dlcd2203) [reply libs/sdk-backend-tiger/src/backend/organization/settings.ts:37]: Because some settings has Dynamic ID (ex: Whitelabeling), so we should include the type of setting to result of function. After that, we can filter it on UI.
Screen Shot 2023-04-03 at 16 08 56

File: libs/sdk-backend-spi/src/organization/settings/index.ts (hunk 4 above)
Comment (dlcd2203) [reply libs/sdk-backend-spi/src/organization/settings/index.ts]: fixed

File: libs/sdk-backend-spi/src/organization/settings/index.ts (hunk 3 above)
Comment (dlcd2203) [reply libs/sdk-backend-spi/src/organization/settings/index.ts]: fixed

File: libs/sdk-backend-spi/src/organization/settings/index.ts (hunk 2 above)
Comment (dlcd2203) [reply libs/sdk-backend-spi/src/organization/settings/index.ts]: fixed

File: libs/sdk-backend-spi/src/organization/settings/index.ts (hunk 1 above)
Comment (dlcd2203) [reply libs/sdk-backend-spi/src/organization/settings/index.ts]: fixed

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts [hunk 10]
```diff
@@ -18,6 +18,14 @@ export class OrganizationSettingsService
         return this.setSetting("WHITE_LABELING", whiteLabeling);
     }
 
+    public async setTimezone(timezone: string): Promise<void> {
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/organization/settings.ts:21]: It seems that we are missing an implementation of `setTimezone` and `setDateFormat` in [sdk-backend-bear](https://github.com/gooddata/gooddata-ui-sdk/tree/master/libs/sdk-backend-bear), aren't we?

→ Please add such implementation to `sdk-backend-bear`.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts (hunk 9 above)
Comment (dlcd2203) [reply libs/sdk-backend-tiger/src/backend/organization/settings.ts]: I have updated new code at here. Please help me review again

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts (hunk 10 above)
Comment (dlcd2203) [reply libs/sdk-backend-tiger/src/backend/organization/settings.ts:21]: We have already added these functions (setTimezone, setDateFormat) to `sdk-backend-base `, so we don't need to do it in Bear.

**Review by anvuong119 — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12365/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14294/ : SUCCESS in 15m 06s

**Comment by anvuong119** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12369/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14298/ : SUCCESS in 15m 01s

File: common/changes/@gooddata/sdk-ui-all/dedo-sd-2656_2023-04-06-14-31.json [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@gooddata/sdk-ui-all",
+      "comment": "sdk-ui-backend-tiger: add apis to config localization, timezoe for Org",
```
Comment (anvuong119) [inline common/changes/@gooddata/sdk-ui-all/dedo-sd-2656_2023-04-06-14-31.json:5]: There is a typo here, it should be `timezone` not `timezoe`, shouldn't it?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12370/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14299/ : SUCCESS in 15m 09s

File: libs/sdk-backend-tiger/src/backend/organization/settings.ts [hunk 12]
```diff
@@ -18,15 +18,24 @@ export class OrganizationSettingsService
...
+        return this.setSetting("FORMAT_LOCALE", { value: dateFormat });
+    }
+
     public async getSettings(): Promise<ISettings> {
         const { data } = await this.authCall(async (client) =>
             client.entities.getAllEntitiesOrganizationSettings({}),
         );
 
         return data.data.reduce((result: ISettings, setting) => {
+            const id = setting.attributes?.type === "WHITE_LABELING" ? "whiteLabeling" : setting.id;
```
Comment (hackerstanislav) [inline libs/sdk-backend-tiger/src/backend/organization/settings.ts:35]: Sry that i enter into discussion so alte, but this is not good approach. Every setting has dynamic id now on tiger, not only white labeling, so basically we need to this mapping based on type, but not in SDK. This must be done on client side or introduce new api that return settings mapped by type, this was a public api change, that is wrong.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12544/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14438/ : SUCCESS in 15m 14s

**Review by anvuong119 — APPROVED** [review]

**Comment by anvuong119** [comment]
LGTM

**Comment by anvuong119** [comment]
As changes in `libs/sdk-backend-tiger/src/backend/organization/settings.ts` have been done by @hackerstanislav , this PR is mergeable.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3611

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2605/ : SUCCESS in 10m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/487/ : SUCCESS in 1m 25s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14440/ : SUCCESS in 15m 30s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/68/ : SUCCESS in 10m 02s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/102/ : UNSTABLE in 12m 44s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3761/ : SUCCESS in 16s
